### ISSUE TITLE
Missing rows and inability to query with group by clause

### ISSUE THREAD

**Issue by PeterOsinski** [jtarchie/sqlitezstd#17]
Hi, thanks for spending the time on the project. I gave it a try and I realized that the compressed database (using `go run github.com/SaveTheRbtz/zstd-seekable-format-go/cmd/zstdseek`) contains less rows than it's original (I checked with a simple identical count(*) query on both databases).

In addition, group by queries doesn't return any results.
Are you able to backtrack that issue?

**Comment by jtarchie** [comment]
I have tested this against a rather large file: 50GB (uncompressed) and 17GB (compressed). The compression could cause rows to be lost because the file's data integrity has been compromised. I'm assuming that sqlite has mechanisms for this.

However, let's hunt this down. 😃 

Can you provide information about the database size, schema, amount of expected rows, etc?

My to-do is to write a test here to verify that it works properly.

**Comment by jtarchie** [comment]
I've added a [test for data integrity](https://github.com/jtarchie/sqlitezstd/commit/e0b7f7ec62054bd5d182b96351d4f9127abb71d2). It passes, let's check that its what you are looking for, however.

I'll write something to double-check my 50GB and 17GB files, too. To ensure it is all working.

**Comment by jtarchie** [comment]
After further investigating the test, I upped it from 10k elements to 100k elements. I do receive an error, which isn't the one you pointed out, but it seems interesting. 🤔

**Comment by jtarchie** [comment]
I think the issue I am having with 10k to 100k elements is actually with the library I use to scan for results. I added a `COUNT(*)` in there and it seems to work just fine.

**Comment by jtarchie** [comment]
> Thanks for taking a look at it, I'll try to reproduce this issue with a minimal setup and provide it.

Thank you. I'm happy to pair on it if that would help. I'm able to recreate the issue. That doesn't mean it isn't happening; I don't know the correct incantation to reproduce it.

**Comment by jtarchie** [comment]
I checked against my 50GB uncompressed and 17GB compressed. They contain the same data, based on a `SELECT COUNT(*)` of over 200+ tables, with the average table having 1 million entries.

The next steps are to ensure the compression is happening correctly on your end.

- `github.com/SaveTheRbtz/zstd-seekable-format-go/cmd/zstdseek` did go through some refactorings recently. A newer version could resolve an unknown issue.
- Ensure that compression is happening in the database with no writers. If data was written while compressing, things could be out of sync.

**Comment by jtarchie** [comment]
Good find!

This looks to be caused by creating a temporary table. I can see the `open` with no filename. When I add `PRAGMA temp_store = memory;` this error does not occur.

Working on the solution.

**Comment by jtarchie** [comment]
Just capturing some notes:

In this [example VFS](https://www.sqlite.org/src/doc/trunk/src/test_demovfs.c) they note:

```
**     3. Temporary files. The user must configure SQLite to use in-memory
**        temp files when using this VFS. The easiest way to do this is to
**        compile with:
**
**          -DSQLITE_TEMP_STORE=3
```

I wonder if this is one of those if you are going to use VFSes you have to worry about this.

**Comment by jtarchie** [comment]
With the current implementation, I don't have access to bypass the VFS layer to the default.

The best method to work around is set `PRAGMA temp_store = memory;`.

I'll add this to the README.

**Comment by jtarchie** [comment]
I have verified with your example this change works.

**Comment by jtarchie** [comment]
I'll reopen if I cannot resolve with https://github.com/psanford/sqlite3vfs/issues/15

**Comment by jtarchie** [comment]
🤦 Yeah, that makes sense. `sql` returns from a pool, so the pragma doesn't necessarily apply to all clients.
I'll fix the example.

**Comment by jtarchie** [comment]
Updated the docs. Thanks for discovering this and providing a helpful example to narrow it down.

1. I was able to fix this in the C version of the extension I'm maintaining. I wrote it because I realized (like this) that issues may come up, which would be easier in C than with CGO. That extension is private for potential licensing reasons, which I am discussing with some users now.
2. This work around and documentation will be helpful for future people. We've documented it well here, the README, and now have a test case. We've done our due diligence.
